pmd-cli/src/main/java/net/sourceforge/pmd/cli/commands/internal/PmdCommand.java

```diff
@@ -140,9 +143,20 @@ public void setBenchmark(final boolean benchmark) {
         this.benchmark = benchmark;
     }
 
-    @Option(names = "--short-names", description = "Prints shortened filenames in the report.")
-    public void setShortnames(final boolean shortnames) {
-        this.shortnames = shortnames;
+    @Option(names = { "--relativize-paths-with", "-z"}, description = "Path relative to which directories are rendered in the report. "
+            + "This option allows shortening directories in the report; "
+            + "without it, paths are rendered as mentioned in the source directory (option \"--dir\"). "
+            + "The option can be repeated, in which case the shortest relative path will be used. "
+            + "If the root path is mentioned (e.g. \"/\" or \"C:\\\"), then the paths will be rendered as absolute.")
+    public void setRelativizePathsWith(List<String> rootPaths) {
```

Comment:
unlike JCommander, PicoCli has first class support for `Path`, you can ask for `List<Path>` directly.

Reply:
+1